src/goto-symex/symex_heuristic_transformation.h

```diff
@@ -0,0 +1,23 @@
+// Copyright 2016-2019 Diffblue Limited. All Rights Reserved.
+
+#ifndef CPROVER_GOTO_SYMEX_SYMEX_HEURISTIC_TRANSFORMATION_H
+#define CPROVER_GOTO_SYMEX_SYMEX_HEURISTIC_TRANSFORMATION_H
+
+#include "complexity_violation.h"
+#include "symex_complexity.h"
+#include <util/std_expr.h>
+
+/// Default heuristic transformation that cancels branches when complexity
+/// has been breached.
+class symex_heuristic_transformationt
```

Comment:
I'd prefer to make this class abstract, to show it is an interface, or even replace it by `std::function<void(const complexity_violationt, goto_symex_statet &)>`